src/clr-angular/utils/id-generator/id-generator.spec.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Component, Inject, NgModule, Type} from "@angular/core";
+import {ComponentFixture, TestBed} from "@angular/core/testing";
+import {By} from "@angular/platform-browser";
+
+import {UNIQUE_ID, UNIQUE_ID_PROVIDER} from "./id-generator.service";
+
+@Component({
+    selector: "test-zero",
+    template: `
+        <div [id]="divId">My Div</div>`,
+    providers: [UNIQUE_ID_PROVIDER]
+})
+class IdZero {
+    constructor(@Inject(UNIQUE_ID) public divId: string) {}
+}
+
+@Component({
+    selector: "test-one",
+    template: `
+        <div [id]="divId">My Div</div>`,
+    providers: [UNIQUE_ID_PROVIDER]
+})
+class IdOne {
+    constructor(@Inject(UNIQUE_ID) public divId: string) {}
+}
+
+@Component({
+    selector: "test-two",
+    template: `
+        <div [id]="divId">My Div</div>`,
+    providers: [UNIQUE_ID_PROVIDER]
+})
+class IdTwo {
+    constructor(@Inject(UNIQUE_ID) public divId: string) {}
+}
+
+@NgModule({declarations: [IdZero, IdOne, IdTwo], exports: [IdZero, IdOne, IdTwo]})
+class IdTestingModule {}
+
+@Component({
+    template: `
+        <test-zero></test-zero>
+        <test-one></test-one>
+        <test-two></test-two>`
```

Comment:
Why do you need 3 different components when they're exact clones of one another?
Just use the same one 3 times, and when querying just iterate over the array of 3 elements to check the ids.

Reply:
Because more is better? j/k Duh, not sure what I was doing with all three. Removed.